src/current/v23.2/cluster-setup-troubleshooting.md

```diff
@@ -174,11 +174,17 @@ The `kv.snapshot_rebalance.max_rate` and `kv.snapshot_recovery.max_rate` [cluste
 
 However, if the settings are too high when nodes are added to the cluster, this can cause degraded performance and node crashes. We recommend **not** increasing these values by more than 2 times their [default values]({% link {{ page.version.version }}/cluster-settings.md %}) without explicit approval from Cockroach Labs.
 
-**Explanation:** If `kv.snapshot_rebalance.max_rate` and `kv.snapshot_recovery.max_rate` are set too high for the cluster during scaling, this can cause nodes to experience ingestions faster than compactions can keep up, and result in an [inverted LSM]({% link {{ page.version.version }}/architecture/storage-layer.md %}#inverted-lsms).
+**Explanation:** If `kv.snapshot_rebalance.max_rate` and `kv.snapshot_recovery.max_rate` are set too high for the cluster during scaling, this can cause nodes to experience ingestions faster than [compactions]({% link {{ page.version.version }}/architecture/storage-layer.md %}#compaction) can keep up, and result in an [inverted LSM]({% link {{ page.version.version }}/architecture/storage-layer.md %}#inverted-lsms).
 
 **Solution:** [Check LSM health]({% link {{ page.version.version }}/common-issues-to-monitor.md %}#lsm-health). {% include {{ page.version.version }}/prod-deployment/resolution-inverted-lsm.md %}
 
-After compaction has completed, lower `kv.snapshot_rebalance.max_rate` and `kv.snapshot_recovery.max_rate` to their [default values]({% link {{ page.version.version }}/cluster-settings.md %}). As you add nodes to the cluster, slowly increase both cluster settings, if desired. This will control the rate of new ingestions for newly added nodes. Meanwhile, monitor the cluster for unhealthy increases in [IOPS]({% link {{ page.version.version }}/common-issues-to-monitor.md %}#disk-iops) and [CPU]({% link {{ page.version.version }}/common-issues-to-monitor.md %}#cpu).
+After [compaction]({% link {{ page.version.version }}/architecture/storage-layer.md %}#compaction) has completed, lower `kv.snapshot_rebalance.max_rate` and `kv.snapshot_recovery.max_rate` to their [default values]({% link {{ page.version.version }}/cluster-settings.md %}). As you add nodes to the cluster, slowly increase both cluster settings, if desired. This will control the rate of new ingestions for newly added nodes. Meanwhile, monitor the cluster for unhealthy increases in [IOPS]({% link {{ page.version.version }}/common-issues-to-monitor.md %}#disk-iops) and [CPU]({% link {{ page.version.version }}/common-issues-to-monitor.md %}#cpu).
+
+{{site.data.alerts.callout_success}}
+[XXX](): <font color="red">QUESTION</font>: the instructions above say to reduce your workload concurrency, so it seems like maybe we should NOT be recommending increasing compaction concurrency here? b/c if we had IOPs and CPU headroom we would not be in this bad state, and thus increasing compaction concurrency would use *even more* IO / CPU right?  That's my sense anyway but please confirm.
```

Comment:
If we're falling behind, there's two possibilities:

1. Hardware resources (IO or CPU) are exhausted.
2. Compaction concurrency is exhausted.

It can be hard to differentiate the two, but it's possible for rebalancing to grow the LSM too fast while still not exhausting IO or CPU, because in order to fully utilize the resources we need more concurrent compactions.

To complicate things a little more, once the LSM becomes inverted, you may see high CPU usage as a consequence of the inverted LSM. But it might be the case that if you had increased the compaction concurrency earlier, the LSM may have never become inverted, and CPU usage may have remained within acceptable levels.

I'm not sure what all that means with respect to what we should document.

Reply:
Ya what you describe does seem quite complicated to understand, especially if one is in the middle of following these other instructions that seem to be about "KV things"

I wonder if adding this info here adds too much complexity / branchiness on top of the KV snapshots stuff and is not really actionable enough 

Maybe it's enough in this section just to add the link to the 'Storage Layer > compactions' since that section also has the blurb about this env var and (for now) we can leave it out of here until we have more experience using this env var during escalations / overload situations, at which point we could document how to check "will this help anyway?"

in other words it seems reasonable to start by documenting the variable in more of a "try this out with your setup" context and remove it from this "your stuff is on fire, set this env var and see what happens" context

does that seem reasonable to you?

Reply:
Yeah, that's reasonable. What do you think about also adding a tidbit into the inverted LSM resolution blurb saying that _if you have available CPU and disk bandwidth headroom_, you may consider increasing the compaction concurrency to reshape the LSM faster.

Reply:
> What do you think about also adding a tidbit into the inverted LSM resolution blurb saying that if you have available CPU and disk bandwidth headroom, you may consider increasing the compaction concurrency to reshape the LSM faster.

sounds great! I revised the inverted LSM include file to pull in the info about the new env var.  And then also rewrote the info about the new env var to mention both that it _could_ help during inverted LSMs or be used to increase overall performance (but pls test it etc).  

PTAL and let me know what you think!